pkg/server/server.go

```diff
@@ -81,10 +81,15 @@ func appendEncapsulated(conf *igntypes.Config, mc *mcfgv1.MachineConfig, version
 			return fmt.Errorf("error marshalling Ignition config: %w", err)
 		}
 	} else {
+		v2, err := ctrlcommon.ConvertV3ToV2Ignition(*conf)
+		if err != nil {
+			return err
+		}
 		tmpIgnCfg := ign2types.Config{
 			Ignition: ign2types.Ignition{
 				Version: ign2types.MaxVersion.String(),
 			},
+			Passwd: v2.Passwd,
```

Comment:
This looks like you inject it in the v2 block, but not in the v3 block and...I think what comes into that function is a v3 because it's `igntypes`? 

https://github.com/openshift/machine-config-operator/blob/5bbf77952dc145fe40b13614f6bab78bd12dc9d8/pkg/server/server.go#L57-L60

At least, what comes out is v3: 
```
  "spec": {
    "osImageURL": "registry.build05.ci.openshift.org/ci-ln-pl6n542/stable@sha256:42fef51c3f19f40f5573f1ba6a393c3e8fbe53b78eb977aea957f60b85b1458a",
    "baseOSExtensionsContainerImage": "registry.build05.ci.openshift.org/ci-ln-pl6n542/stable@sha256:7a1b374c2509174b0090cf23db6d02c9c18638749a8f56cccafae735c1638d45",
    "config": {
      "ignition": {
        "version": "3.2.0"
      }
    },
    "kernelArguments": [
      "systemd.unified_cgroup_hierarchy=0",
      "systemd.legacy_systemd_cgroup_controller=1"
    ],
    "extensions": [],
    "fips": false,
    "kernelType": "default"
  }
```

Reply:
:facepalm: I totally see the issue now, my eyes must've skipped over the v3 marshalling in the server...